src/control/cmd/dmg/system.go

```diff
@@ -319,26 +336,30 @@ func (cmd *systemStopCmd) Execute(_ []string) error {
 	req.Hosts.ReplaceSet(hostSet)
 	req.Ranks.ReplaceSet(rankSet)
 
-	// TODO DAOS-5079: group errors when ranks don't exist
 	resp, err := control.SystemStop(context.Background(), cmd.ctlInvoker, req)
+	errCmd := errors.Wrap(err, "System-Stop command")
+	errResp := errors.Wrap(resp.Errors(), "System-Stop command")
 
 	if cmd.jsonOutputEnabled() {
```

Comment:
Seeing how many changes you had to make around this, I think this indicates a need for restructuring. Needn't be done in this PR, but maybe it could if you're feeling it. What I'm thinking of is a cmd.output() method that could encapsulate the JSON check and outputJSON call. Maybe we could pass a closure that formats the output appropriately for the non-JSON case. But that would centralize all the output logic, so we won't miss adding the JSON formatting in the future.

Reply:
I'm not completely sure on what that would look like, I'm working on a follow-up PR that will reorganise dmg/system.go and will look into this.

Reply:
Also to be clear the change was more related to returning an error that reflects any failures in the nested response results which is additional behaviour.

Reply:
About the generic output function, I'm thinking something like:
```
func (cmd *outputCmd) output(data interface{}, func formatter(in interface {}) (string, error)) {
        if jsonCmd, ok := cmd.(jsonOutputter); ok {
                if jsonCmd.jsonOutputEnabled() {
                        return jsonCmd.outputJSON(data)
                }
        }
        result, err := formatter(data)
        // check if formatter failed
        // write result string to cmd.writer
        return nil
}
```
We don't have a generic outputCmd but maybe we should. We'd also need to standardize our output formatter functions to take a single interface and check the type.

Reply:
@mjmac as you wrote the json output code, do you want to go down this route, I'm happy to implement if you want.